src/odemis/acq/stream/_static.py

```diff
@@ -256,52 +256,55 @@ def __init__(self, name, data, *args, **kwargs):
         data = [d.getData() if isinstance(d, model.DataArrayShadow) else d for d in data]
 
         # find positions of each acquisition
-        # tuple of 2 floats -> DataArray: position on SEM -> data
-        self._sempos = {}
+        # tuple of 2 floats + 1 string) -> DataArray: position on SEM + position of polarization analyzer -> data
+        self._pos = {}
         for d in data:
+            if not model.MD_POL_MODE:
+                d.metadata[MD_POL_MODE] = None
             try:
-                self._sempos[d.metadata[MD_POS]] = img.ensure2DImage(d)
+                self._pos[d.metadata[MD_POS] + (d.metadata[MD_POL_MODE],)] = img.ensure2DImage(d)
             except KeyError:
                 logging.info("Skipping DataArray without known position")
 
         # Cached conversion of the CCD image to polar representation
         # TODO: automatically fill it in a background thread
-        self._polar = {} # dict tuple 2 floats -> DataArray
+        self._polar = {}  # dict tuple 2 floats -> DataArray
 
-        # SEM position displayed, (None, None) == no point selected
-        self.point = model.VAEnumerated((None, None),
-                     choices=frozenset([(None, None)] + list(self._sempos.keys())))
+        # SEM position displayed, (None, None, None) == no point selected (x, y, pol)
+        self.point = model.VAEnumerated((None, None, None),
```

Comment:
I don't think it's a good idea to just extend `.point` to a 3rd dimension. First, that would require changes to "all" code using this VA (admittedly, it's not that many, probably just a few places in the GUI). Second, because SEM pos and polarization are unrelated, and so will not likely change simultaneously. The GUI controls would either change the X/Y or the polarization.
=> leave `.point` and create `.polarization`.

Also, add a check that every position has the same set of polarizations. Probably, you should also not add this VA if the only polarization is `None`.